packages/cozy-client/src/associations/HasManyTriggers.js

```diff
@@ -0,0 +1,43 @@
+import get from 'lodash/get'
+
+import HasMany from './HasMany'
+
+const TRIGGERS_DOCTYPE = 'io.cozy.triggers'
+
+/**
+ * Memoize the fetch promise. For now we are only fetching it once but it will
+ * evolve in the future.
+ */
+let memoizedFetchPromise = null
+
+/**
+ * Association used for konnectors to retrieve all their related triggers.
+ * @extends HasMany
+ */
+class HasManyTriggers extends HasMany {
+  /**
+   * In this association the query is special, we need to fetch all the triggers
+   * having for the 'konnector' worker, and then filter them based on their
+   * `message.konnector` attribute
+   */
+  static async query(doc, client) {
+    // Check the memoized promise instead of the redux store to avoid executing
+    // the same query several times (one time per konnector)
+    if (!memoizedFetchPromise) {
+      memoizedFetchPromise = client.query(
+        client.all(TRIGGERS_DOCTYPE).where({ worker: 'konnector' })
+      )
+    }
+
+    await memoizedFetchPromise
```

Comment:
The client can change of Cozy in its lifetime, I would use _.memoize and pass a key function based on the client URI to be asfe.

const getKonnectorTriggers = memoize(client => {
  return client.all(TRIGGERS_DOCTYPE).where({ worker: 'konnector' })
}, client => client.stackClient.uri)

Reply:
Thanks for pointing me this point out, I am too little familiar with mobile constraintes 👍

We will be able to get rid of this memoization when https://github.com/cozy/cozy-client/issues/399 will be implemented.

Reply:
I fixed up the code.